R/read.R

```diff
@@ -1,11 +1,36 @@
 orderly_read <- function(path, call = NULL) {
   entrypoint_filename <- find_entrypoint_filename(path)
-  orderly_read_r(file.path(path, entrypoint_filename), entrypoint_filename)
+  orderly_parse_file(file.path(path, entrypoint_filename))
 }
 
 
-orderly_read_r <- function(path, entrypoint_filename) {
+#' Parse the orderly entrypoint script
+#'
+#' For expert use only.
+#'
+#' Parses details of any calls to the orderly_ in-script functions
+#' into intermediate representation for downstream use. Also validates
+#' that any calls to `orderly_*` in-script functions are well-formed.
+#'
+#' @param path Path to `orderly_*` script
+#'
+#' @return Parsed orderly entrypoint script
+#' @export
+orderly_parse_file <- function(path) {
+  assert_file_exists(path)
   exprs <- parse(file = path)
+  orderly_parse_expr(exprs, basename(path))
+}
+
+
+#' @param exprs Parsed AST from `orderly_*` script
+#' @param filename Name of `orderly_*` file to include in metadata, required if
+#'   calling with `exprs`. If called with a `path` this can be NULL.
```

Comment:
i think might've been from before?
```suggestion
#' @param exprs Parsed AST from `orderly_*` script
#' @param filename Name of `orderly_*` file to include in metadata
```